desafio-05/ancogamer/go/func.go

```diff
@@ -0,0 +1,327 @@
+
+	slice, found := (*bigSalaryByLastName)[dat.Funcs[c].Sobrenome]
+	if found {
+		for _, value := range slice {
+			if dat.Funcs[c].Salario > value.Salario {
+				slice = ([]*Employee{&dat.Funcs[c]})
+				return
+			}
+			if dat.Funcs[c].Salario == value.Salario {
+				slice = append(slice, &dat.Funcs[c])
+				return
+			}
+		}
+		return
+	}
+	(*bigSalaryByLastName)[dat.Funcs[c].Sobrenome] = []*Employee{&dat.Funcs[c]}
+	return
+
+}
+
+func main() {
+	start := time.Now()
+	/*
+		var optCPUProfile string
+		fmt.Println(os.Args[len(os.Args)-1])
+		flag.StringVar(&optCPUProfile, "cpuprofile", "", "write cpu profile to file")
+		flag.Parse()
+		if optCPUProfile != "" {
+			println("to aqui")
+			f, err := os.Create(optCPUProfile)
+			if err != nil {
+				panic(err)
+			}
+			pprof.StartCPUProfile(f)
+			defer pprof.StopCPUProfile()
+		}
+	*/
+	rawdata, err := ioutil.ReadFile(os.Args[1])
+	if err != nil {
+		panic(err)
+	}
+
+	var dat JSON
+	err = json.Unmarshal(rawdata, &dat)
+
+	if err != nil {
+		panic(err)
+	}
+
+	var maiorValor float64 = dat.Funcs[0].Salario
+
+	var menorValor float64 = dat.Funcs[0].Salario
+
+	var maiorArea int
+
+	var mediaGlobalSal float64
+
+	var globalMax, globalMin strings.Builder
+	var mostAndLestEmploy strings.Builder
+
+	var sizeArea int = len(dat.Areas)
+
+	bigSalaryByLastName := lastNameSal{}
+
+	for c = 0; c < len(dat.Funcs); c++ {
+		// Calculo elementos parte 4
+		maxSalByLastName(&bigSalaryByLastName, dat)
+
+		// Parte 1
+		if dat.Funcs[c].Salario > maiorValor {
+			maiorValor = dat.Funcs[c].Salario
+			globalMax.Reset()
+			globalMax.WriteString("global_max|")
+			globalMax.WriteString(dat.Funcs[c].Nome)
+			globalMax.WriteString(" ")
+			globalMax.WriteString(dat.Funcs[c].Sobrenome)
+			globalMax.WriteString("|")
+			globalMax.WriteString(strconv.FormatFloat(math.Round(maiorValor), 'f', 2, 64))
+		} else if dat.Funcs[c].Salario == maiorValor {
+			globalMax.WriteString("\nglobal_max|")
+			globalMax.WriteString(dat.Funcs[c].Nome)
+			globalMax.WriteString(" ")
+			globalMax.WriteString(dat.Funcs[c].Sobrenome)
+			globalMax.WriteString("|")
+			globalMax.WriteString(strconv.FormatFloat(math.Round(maiorValor), 'f', 2, 64))
+		}
+		if dat.Funcs[c].Salario < menorValor {
+			menorValor = dat.Funcs[c].Salario
+			globalMin.Reset()
+			globalMin.WriteString("global_min|")
+			globalMin.WriteString(dat.Funcs[c].Nome)
+			globalMin.WriteString(" ")
+			globalMin.WriteString(dat.Funcs[c].Sobrenome)
+			globalMin.WriteString("|")
+			globalMin.WriteString(strconv.FormatFloat(math.Round(menorValor), 'f', 2, 64))
+		} else if dat.Funcs[c].Salario == menorValor {
+			globalMin.WriteString("\nglobal_min|")
+			globalMin.WriteString(dat.Funcs[c].Nome)
+			globalMin.WriteString(" ")
+			globalMin.WriteString(dat.Funcs[c].Sobrenome)
+			globalMin.WriteString("|")
+			globalMin.WriteString(strconv.FormatFloat(math.Round(menorValor), 'f', 2, 64))
+		}
+
+		mediaGlobalSal += dat.Funcs[c].Salario
+
+		for a := 0; a < sizeArea; a++ {
```

Comment:
WTF is `a`? :) Use um nome mais descritivo.

Reply:
é um contador !

Reply:
resolvido